lib/Parse/ParseDecl.cpp

```diff
@@ -1161,6 +1160,74 @@ Parser::parseDifferentiatingAttribute(SourceLoc atLoc, SourceLoc loc) {
   {
     SyntaxParsingContext ContentContext(
         SyntaxContext, SyntaxKind::DerivativeRegistrationAttributeArguments);
+    // Parse the 'of:' label and colon.
+    if (parseSpecificIdentifier("of", diag::attr_missing_label, "of",
+                                AttrName) ||
+        parseToken(tok::colon, diag::expected_colon_after_label, "of")) {
+      return makeParserError();
+    }
+    {
+      // Parse the name of the function.
+      SyntaxParsingContext FuncDeclNameContext(SyntaxContext,
+                                               SyntaxKind::FunctionDeclName);
+      // NOTE: Use `afterDot = true` and `allowDeinitAndSubscript = true` to
+      // enable, e.g. `@derivative(of: init)` and `@derivative(of: subscript)`.
+      original.Name = parseUnqualifiedDeclName(
+          /*afterDot*/ true, original.Loc,
+          diag::attr_derivative_expected_original_name, /*allowOperators*/ true,
+          /*allowZeroArgCompoundNames*/ true, /*allowDeinitAndSubscript*/ true);
+      if (consumeIfTrailingComma())
+        return makeParserError();
+    }
+    // Parse the optional 'wrt' differentiation parameters clause.
+    if (isIdentifier(Tok, "wrt") &&
+        parseDifferentiationParametersClause(params, AttrName))
+      return makeParserError();
+  }
+  // Parse ')'.
+  if (!consumeIf(tok::r_paren, rParenLoc)) {
+    diagnose(getEndOfPreviousLoc(), diag::attr_expected_rparen, AttrName,
+             /*DeclModifier*/ false);
+    return makeParserError();
+  }
+  return ParserResult<DerivativeAttr>(
+      DerivativeAttr::create(Context, /*implicit*/ false, atLoc,
+                             SourceRange(loc, rParenLoc), original, params));
+}
+
+/// SWIFT_ENABLE_TENSORFLOW
+ParserResult<DerivativeAttr>
+Parser::parseDifferentiatingAttribute(SourceLoc atLoc, SourceLoc loc) {
```

Comment:
Note: I duplicated `parseDifferentiatingAttribute` rather than adding a flag to `parseDerivativeAttribute` to be future-facing. It's easier to remove duplicated code.

I avoided unnecessary/duplicate code in other places (e.g. not defining a new `DifferentiatingAttr` class).